simpleviewinc/keg-herkin
# Sidebar update

## Context

* Update the sidebar treeList style to match the style of other list component

## Goal

* Update the styling of tree List component
* add styling to be able to distinguish which folder/file is active/inactive

## Updates
* `repos/tap/components/list/treeList/treeList.js`
  * updated jsdoc slightly
  * update the NodeComponent to pull the styles based on current state

* `repos/tap/theme/components/sidebar/sideBarContent.js`
  * updated sidebar header height to match the height of modal header  

* `repos/tap/theme/components/treeList/treeList.js`
 * added styles based on state
 
## Testing

* pull this branch
* run `keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:sidebar-update`
* navigate to http://herkin-sidebar-update.local.kegdev.xyz/
* open the sidebar
* Verify on hover, text change to primary color (blue-ish)
* Verify on active folder, text keeps the primary color
* Verify on file hover, text is success color (green-ish)
  * [image](https://user-images.githubusercontent.com/3317835/108607505-044a8c00-737e-11eb-9ec1-dc876197cad4.png)
* click on any file to make them active
* Verify when a file is active, it remains highlighted
  * [image](https://user-images.githubusercontent.com/3317835/108607533-217f5a80-737e-11eb-83c3-5bbdb0f453ef.png)
